src/regmod/data.py

```diff
@@ -11,7 +11,85 @@
 
 @dataclass
 class Data:
-    col_obs: Union[str, List[str]] = None
+    """Data class used for validate and accessing data in `pd.DataFrame`.
+
+    Parameters
+    ----------
+    col_obs : Optional[Union[str, List[str]]], optional
+        Column name(s) for observation. Default to be `None`.
+    col_covs : List[str], optional
+        Column names for covariates. Default to be an empty list.
+    col_weights : str, default="weights"
+        Column name for weights. Default to be `'weights'`. If `col_weights` is
+        not in the data frame, a column with name `col_weights` will be added to
+        the data frame filled with 1.
+    col_offset : str, default="offset"
+        Column name for weights. Default to be `'offset'`. If `col_offset`
+        is not in the data frame, a column with name `col_offset` will be added
+        to the data frame filled with 0.
+    df : pd.DataFrame, optional
+        Data frame for the object. Default is an empty data frame.
+
+    Attributes
+    ----------
+    obs
+    covs
+    weights
+    offset
+    trim_weights
+    num_obs
+    col_obs : Optional[Union[str, List[str]]]
+        Column name for observation, can be a single string, a list of string or
+        `None`. When it is `None` you cannot access property `obs`.
+    col_covs : List[str]
+        A list of column names for covariates.
+    col_weights : str
+        Column name for weights. `weights` can be used in the likelihood
+        computation. Values of `weights` are required be between 0 and 1.
+        `col_weights` defaultly is set to be `'weights'`. If `col_weights` is
+        not in the data frame, a column with name `col_weights` will be added to
+        the data frame filled with 1.
+    col_offset : str
+        Column name for offset. Same as `weights`, `offset` can be used in
+        computing likelihood. `offset` need to be pre-transformed according to
```

Comment:
"`offset` needs to be pre-transformed according to the link function of the parameters. Default for `col_offset` is `'offset'`."